Lib/ntpath.py

```diff
@@ -816,21 +837,19 @@ def commonpath(paths):
         curdir = '.'
 
     try:
-        drivesplits = [splitdrive(p.replace(altsep, sep).lower()) for p in paths]
-        split_paths = [p.split(sep) for d, p in drivesplits]
+        drivesplits = [splitroot(p.replace(altsep, sep).lower()) for p in paths]
+        split_paths = [p.split(sep) for d, r, p in drivesplits]
 
-        try:
-            isabs, = set(p[:1] == sep for d, p in drivesplits)
-        except ValueError:
-            raise ValueError("Can't mix absolute and relative paths") from None
+        if len(set(r for d, r, p in drivesplits)) != 1:
```

Comment:
Since we're touching this line anyway, we may as well modernise the code slightly by using a set comprehension

```suggestion
        if len({r for d, r, p in drivesplits}) != 1:
```